2996_reflection/reflection.md

```diff
@@ -3576,12 +3545,22 @@ Add a new paragraph between [expr.eq]{.sref}/5 and /6:
 
 ### [expr.const]{.sref} Constant Expressions {-}
 
-Add a new paragraph after the definition of _manifestly constant-evaluated_ [expr.const]{.sref}/20:
+Add a new paragraph after the definition of _potentially constant-evaluated_ [expr.const]{.sref}/21:
 
 ::: std
 ::: addu
 
-[21]{.pnum} An expression or conversion is _plainly constant-evaluated_ if it is:
+[22]{.pnum} The _evaluation context_ is a set of points within the program that determines which declarations are visible to certain expressions used for reflection. During the evaluation of a set of manifestly constant-evaluated expressions `$M$`, let `$E$` be the unique such expression whose result is required for the evaluation of every other expression in `$M$`. The evaluation context comprises the points in the instantiation context of `$E$` ([module.context]).
```

Comment:
I don't entirely understand what's going on here. When do we evaluate a "set" of manifestly constant-evaluated expressions? like
```cpp
if constexpr (E1) { ... } 
else if constexpr (E2) { ... }
```
`E1` and `E2` are two different manifestly constant evaluated expressions, what is the unique expression whose result is required for both of these?

Reply:
Re: a "set" of manifestly constant-evaluated expressions:

```cpp
constexpr auto r = [] {
  static constexpr auto s = members_of(^Cls).size();
    // During evaluation of initializer of 's', 'r' is also still being evaluated.
    //
    // The evaluation context would be the program point of 'members_of', not the
    // program point of the lambda.
    
  return s;
};
```

Re: your example above - `E1` is plainly constant evaluated, so `E2` must be parsed in a context in which `E1` has already been evaluated.

(hopefully that helps illustrate what I'm trying to describe - the program point of the expression under evaluation that's "furthest down the stack").

Reply:
I really have no idea what this means. Definitely could use a few examples? 

Nor do I even understand the `members_of` wording:
> A `vector` containing reflections of all _members-of-representable_ members of the entity represented by `r` that are _members-of-visible_ from one of the points in the evaluation context ([expr.const]).

Which likewise should probably add some examples of cases where some member is not visible.

Reply:
Is this dropping the definition of "plainly constant evaluated"?

Reply:
> Is this dropping the definition of "plainly constant-evaluated"?

No! I hope not 😅 It's just inserting prior to "plainly constant evaluated", since it's more related to "manifestly constant-evaluated" than it is to "plainly constant-evaluated".